apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/options/OpenStackSwiftPutOptions.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.swift.v1.options;
+
+import com.google.common.util.concurrent.ListeningExecutorService;
+import org.jclouds.blobstore.options.PutOptions;
+
+public class OpenStackSwiftPutOptions extends PutOptions {
```

Comment:
No, you still don't understand the difference between the provider API and the portable abstraction.  Go compare something like B2 with its [MultipartApi](https://github.com/jclouds/jclouds/blob/master/providers/b2/src/main/java/org/jclouds/b2/features/MultipartApi.java) against [B2BlobStore](https://github.com/jclouds/jclouds/blob/master/providers/b2/src/main/java/org/jclouds/b2/blobstore/B2BlobStore.java).  The former implements the REST API while the latter conforms to the `BlobStore` interface.  This pull request should not contain any of the latter.